limitador/src/storage/infinispan/sets.rs

```diff
@@ -38,15 +38,48 @@ pub async fn add(
 ) -> Result<(), StorageErr> {
     let cache_name = cache_name.into();
     let set_key = set_key.into();
+    let element = element.into();
 
     dist_lock::lock(infinispan, &cache_name, &set_key).await?;
 
+    let res = add_set(infinispan, cache_name.clone(), set_key.clone(), element).await;
+
+    dist_lock::release(infinispan, &cache_name, &set_key).await?;
```

Comment:
:thinking: :thought_balloon: how can I distinguish whether the `add_set` failed or the `release` failed? I think we want to distinguish the cases, because `dist_lock::release`-failure can be a soft-fail with no retry needed, whereas `add_set`-failure would be much more like a real, problematic failure.

Reply:
I think both need retries. The difference is that if after retrying `add_set` still fails, we should send the error to the user whereas with `release` we shouldn't. Actually, `release` does not even need to run on request time.
Given that we don't have the logic for retries yet, not even in the other redis implementations, I suggest we tackle this on a future PR.

Reply:
But adding to a set being successful might not need a retry (if only it might be noted down that the lock could not be successfully released), no?

Reply:
Ok with me to tackle it in a future PR.